### PR TITLE
[proposed spec change] subscribe to whole subtree of container

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by michielbdejong** [comment]
no need to review this now, just documenting that this is something we're experimenting with in https://github.com/inrupt/websockets-pubsub

**Comment by michielbdejong** [comment]
See also #180

**Comment by michielbdejong** [comment]
Do not merge unless NSS also implements this.

File: api-websockets.md [hunk 1]
```diff
@@ -52,9 +52,9 @@ pub https://example.org/data/test
 Only absolute URIs should be used in both the `sub` and the `pub` message.
 
 Subscribing to a container can also be really useful, since all CRUD operations
-(POST, PUT, PATCH, DELETE) performed on resources of that container will trigger
-a notification for the container URI. This makes synchronization between
-multiple apps really easy.
+(POST, PUT, PATCH, DELETE) performed on resources of that container and its subcontainers will trigger
```
Comment (RubenVerborgh) [inline api-websockets.md]: Recursively or not?

Comment (michielbdejong) [reply api-websockets.md]: yes, sorry.

**Comment by RubenVerborgh** [comment]
Thanks for clarifying the recursive part, @michielbdejong.

At this point, I want to express my concern regarding the number of notifications that will nee to be generated by the server, and processed by the receiver.

If we want recursive notifications, we might want to make such recursive subscriptions a) explicitly opt-in (so non-recursive is also still possible) and/or b) optional for the server.

**Comment by michielbdejong** [comment]
Yes, I see your point. But note that notifications will only be sent for resources that the app has read access to.

And even without that change, it's not safe to assume that the number notifications to be sent will not be prohibitive, so this will generally increases the number in practice, but the worst-case number (number of resources in a single folder that the client is subscribed to) was already huge.

**Comment by RubenVerborgh** [comment]
> But note that notifications will only be sent for resources that the app has read access to.

That doesn't necessarily make it better for the server though, because now every change within a subscribed subfolder should be ran through the ACL process.

> the number notifications to be sent (number of resources in a single container that the client is subscribed to) might be prohibitive, so this will generally increases the number in practice, but the worst-case number was already huge.

Yes, but it used to be the case that you needed to do a lot of actions to get a large volume; now a single action can get you a large volume. That works both ways.

This reinforces my belief that we should _design_ an update subscription mechanism instead of extending a quick hack that happened to be built at some point (https://github.com/solid/solid-spec/pull/179#issuecomment-497297385).

**Comment by michielbdejong** [comment]
> every change within a subscribed subfolder should be ran through the ACL process.

that's already the case right? also if container subscription is not recursive?

> design a proper notification mechanism from scratch

Sure, go ahead and propose something! Let me know if you want input. This PR just documents what inrupt/pod-server currently does.

**Pull request closed without merging**